src/components/global/ModalsStudent.tsx

```diff
@@ -0,0 +1,99 @@
+// import CloseIcon from '@mui/icons-material/Close'
+import OpenInNewIcon from '@mui/icons-material/OpenInNew'
+import { Box, Button, IconButton, Modal, Typography } from '@mui/material'
+import { useState } from 'react'
+import { Classroom, Player, Repeatable, TaskWithStatus } from '../../types'
+import { completeRepeatable, completeTask } from '../../utils/mutations'
+import { format, fromUnixTime } from 'date-fns'
+import { StudentTaskModalBox, ModalTitle, StudentBoxInModal } from '../global/TaskModalStyles'
+
+interface PropsTask {
+	classroom: Classroom
+	player: Player
+	task: TaskWithStatus
+	type: 'task'
+}
+
+interface PropsRepeatables {
+	classroom: Classroom
+	player: Player
+	task: Repeatable
+	type: 'repeatables'
+}
+
+export default function ModalsStudent(props: PropsTask | PropsRepeatables) {
+	// State variables
```

Comment:
Thanks for the comments, but you generally don't need to leave comments for simple stuff like this. Well-written code comments itself!!

Highly recommend watching [this video](https://www.youtube.com/watch?v=Bf7vDBBOBUA)